Add documentation for React Native Privacy Overrides (#29491)
DataDog/documentation: content/en/real_user_monitoring/session_replay/mobile/privacy_options.md

To ensure overrides are recognized properly, they should be applied as early as possible in the view lifecycle. This prevents scenarios where Session Replay might process a view before applying the overrides.

Privacy overrides affect views and their descendants. This means that even if an override is applied to a view where it might have no immediate effect (for example, applying an image override to a text input), the override still applies to all child views.

> Overrides operate using a "nearest parent" principle: if a view has an override, it uses that setting. Otherwise, it inherits the privacy level from the closest parent in the hierarchy with an override. If no parent has an override, the view defaults to the application's general masking level.

Reviewer: ```suggestion
Overrides operate using a "nearest parent" principle: if a view has an override, it uses that setting. Otherwise, it inherits the privacy level from the closest parent in the hierarchy with an override. If no parent has an override, the view defaults to the application's general masking level.

The sections below describe how to override privacy settings for Android and iOS. For settings specific to React Native, see [Privacy overrides support in React Native](#privacy-overrides-support-in-react-native).
```

Use `setSessionReplayHidden(hide = true)` to hide the element. Setting `hide` to `false` removes the override.

{{< code-block lang="kotlin" filename="application.kt" disable_copy="false" collapsible="true" >}}
// Mark a view as hidden
myView.setSessionReplayHidden(hide = true)
// Remove the override from the view
myView.setSessionReplayHidden(hide = false)
{{< /code-block >}}

{{% /tab %}}

{{% tab "iOS" %}}
{{< code-block lang="swift" filename="AppDelegate.swift" disable_copy="false" collapsible="true" >}}
// Mark a view as hidden
myView.dd.sessionReplayOverrides.hide = true
// Remove the override from the view
myView.dd.sessionReplayOverrides.hide = false
{{< /code-block >}}

**Note**: Setting the `hidden` override to `nil` has the same effect as setting it to `false`—it disables the override.

{{% /tab %}}
{{< /tabs >}}

> ### Privacy Overrides Support in React Native

Reviewer: Since it's a different process anyway, let's raise the topic hierarchy for this entire seciton

> Fine-Grained Masking is fully supported in React Native starting from version `2`.`8`.`0` of Datadog's <a href="https://github.com/DataDog/dd-sdk-reactnative">React Native SDK</a>. Although the underlying functionality is shared with native Android and iOS platforms, the integration in React Native is designed to align with common React patterns.

Reviewer: ```suggestion
Fine-grained masking is fully supported in React Native starting from version `2`.`8`.`0` of the Datadog [React Native SDK][2]. Although the underlying functionality is shared with native Android and iOS platforms, the integration in React Native is designed to align with common React patterns.
```

> #### Behavior Consistency

Reviewer: ```suggestion
### Behavior consistency
```
Our style guide calls for sentence capitalization in our headings.

> Internally, React Native uses the same primitives as Android and iOS. Therefore, you can expect identical behavior across platforms when applying privacy settings.

Reviewer: ```suggestion
React Native's implementation is built on the same foundation as the native Android and iOS SDKs. As a result, you can rely on the privacy features behaving the same way across all three platforms.
```

> #### Usage via SessionReplayView

Reviewer: ```suggestion
### Usage using `SessionReplayView`
```

Author: "Usage using" does not sound right

Reviewer: updated suggestion

| `SessionReplayView.Hide`     | Completely hides all child components from session replay.                                                    | _(No additional properties)_                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |

> #### Integration Approaches

Reviewer: ```suggestion
### Integration Approaches
```

Author: Why are we changing this ? They are meant to be all on the same level

{{% /tab %}}
{{< /tabs >}}

> ### Privacy overrides Support in React Native

Reviewer: ```suggestion
### Privacy overrides support in React Native
```

| `SessionReplayView.Hide`     | Completely hides all child components from session replay.                                                    | _(No additional properties)_                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                     |

> #### Integration Approaches

Reviewer: ```suggestion
#### Integration approaches
```
